### PR TITLE
CEDS-3671 Use value of /departure-transport for /border-transport when the latter has no value

### PR DESCRIPTION
**Pull request by udinobi**

### CODE REVIEW

**Review by darrylmurphy — APPROVED** [review]

File: app/uk/gov/hmrc/exports/services/mapping/declaration/BorderTransportMeansBuilder.scala [hunk 1]
```diff
@@ -25,37 +25,39 @@ import wco.datamodel.wco.declaration_ds.dms._2._
...
-      }
+    val hasDepartureTransport = transport.hasDepartureTransportDetails
+
+    if (hasTransportLeavingTheBorder || hasBorder || hasDepartureTransport) {
+      val transportMeans = new Declaration.BorderTransportMeans
+
+      if (hasTransportLeavingTheBorder) appendTransportLeavingTheBorder(transport, transportMeans)
+
+      if (hasBorder) appendBorderTransport(transport, transportMeans)
+      else if (hasDepartureTransport) appendDepartureTransport(transport, transportMeans)
```
Comment (glidester) [inline app/uk/gov/hmrc/exports/services/mapping/declaration/BorderTransportMeansBuilder.scala:40]: This bit seems a little confusing as both `appendBorderTransport` and `appendDepartureTransport` actually sets both BorderTransport and DepartureTransport values when called (but this `if` makes it seems like they are mutually exclusive).

Comment (udinobi) [reply app/uk/gov/hmrc/exports/services/mapping/declaration/BorderTransportMeansBuilder.scala:40]: They both set BorderTransport only, but `appendDepartureTransport` is called **only** if the user did not enter any value in BorderTransport (and in this case `appendBorderTransport` is not called). Note that is Builder is dedicated to BorderTransport only.

File: app/uk/gov/hmrc/exports/services/mapping/declaration/BorderTransportMeansBuilder.scala [hunk 2]
```diff
@@ -25,37 +25,39 @@ import wco.datamodel.wco.declaration_ds.dms._2._
...
-    if (hasDeparture || hasBorder) {
-      if (hasDeparture) {
-        appendDepartureTransport(transport, transportMeans)
-      }
-      if (hasBorder) {
-        appendBorderTransport(transport, transportMeans)
-      }
+    val hasDepartureTransport = transport.hasDepartureTransportDetails
+
+    if (hasTransportLeavingTheBorder || hasBorder || hasDepartureTransport) {
```
[the remark this answers is not in the document]
Comment (udinobi) [reply app/uk/gov/hmrc/exports/services/mapping/declaration/BorderTransportMeansBuilder.scala:34]: But you want exec line 42 only if one these values are defined.

File: app/uk/gov/hmrc/exports/services/mapping/declaration/BorderTransportMeansBuilder.scala (hunk 1 above)
Comment (glidester) [reply app/uk/gov/hmrc/exports/services/mapping/declaration/BorderTransportMeansBuilder.scala:40]: Ok, I see. Maybe `appendDepartureTransport` could be named `appendBorderTransportWithDepartureValue' to make this more obvious?

**Pull request merged**